lib/_tls_common.js

```diff
@@ -146,6 +146,10 @@ exports.createSecureContext = function createSecureContext(options, context) {
     c.context.setFreeListLength(0);
   }
 
+  if (options.clientCertEngine) {
+    c.context.setClientCertEngine(options.clientCertEngine);
```

Comment:
Can you add something like the following?
```js
if (options.clientCertEngine) {
  if (c.context.setClientCertEngine)
    c.context.setClientCertEngine(options.clientCertEngine);
  else
    throw new Error('Custom engines not supported by this OpenSSL.');
}
```
That's a bit nicer than an 'undefined is not a function' TypeError.

Suggestions for a better error message welcome, it's still a bit opaque, perhaps.